extension/sigv4authextension/extension.go

```diff
@@ -87,3 +87,32 @@ func getCredsProviderFromConfig(cfg *Config) (*aws.CredentialsProvider, error) {
 
 	return &awscfg.Credentials, nil
 }
+
+func getCredsProviderFromWebIdentityConfig(cfg *Config) (*aws.CredentialsProvider, error) {
+	tokenRetriever := stscreds.IdentityTokenRetriever(
+		stscreds.IdentityTokenFile(cfg.AssumeRole.WebIdentityTokenFile),
+	)
+	_, err := tokenRetriever.GetIdentityToken()
+	if err != nil {
+		return nil, err
```

Comment:
Please augment errors with useful information.

```suggestion
		return nil, fmt.Errorf("unable to retrieve token file: %w", err)
```